src/plugins/vis_type_vega/public/expressions/line_vega_spec_fn.test.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { cleanString, createSpecFromDatatable, formatDataTable } from './line_vega_spec_fn';
+
+describe('cleanString()', function () {
+  it('string should not contain "', function () {
+    const dirtyString = '"someString"';
+    expect(cleanString(dirtyString)).toBe('someString');
+  });
+});
+
+describe('createSpecFromDatatable()', function () {
+  it('build simple line chart"', function () {
```

Comment:
not trying to block, but I feel we should have some more test cases for this. I get that a lot of the invalid specs will be filtered out at the eligibility layer, but having some tests for empty data, single data point, etc. could be helpful

Might be worth extracting the table generation in a helper fn. Then can call that to generate the table and inject to simplify and shorten the test files

Reply:
if you feel more of this is covered in eligibility checks and integration tests, I'm fine with that too

Reply:
Added a bunch more and broke up the bigger function to make it easier to test different scenarios